src/testing/systest/supervisor/logged_process.zig

```diff
@@ -0,0 +1,299 @@
+//! Runs a subprocess inheriting its stderr output. The processes can terminate by its own or be
+//! actively terminated with a SIGKILL.
+//!
+//! We use this to run a cluster of replicas and the workload.
+//!
+//! NOTE: some duplication exists between this and `tmp_tigerbeetle.zig` that could perhaps be
+//! unified.
+const std = @import("std");
+const builtin = @import("builtin");
+
+const log = std.log.scoped(.logged_process);
+
+const assert = std.debug.assert;
+
+const Self = @This();
+pub const State = enum(u8) { running, terminated, completed };
+const AtomicState = std.atomic.Value(State);
+const Options = struct { env: ?*const std.process.EnvMap = null };
+
+// Passed in to init
+allocator: std.mem.Allocator,
+argv: []const []const u8,
```

Comment:
Let's avoid storing argv. We aren't actually using it anywhere, and, if we don't store it, then the caller wouldn't have to dupe it!